src/OpenApi/src/Comparers/ComparerHelpers.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class ComparerHelpers
+{
+    internal static bool DictionaryEquals<TKey, TValue>(IDictionary<TKey, TValue> x, IDictionary<TKey, TValue> y, IEqualityComparer<TValue> comparer)
+        where TKey : notnull
+        where TValue : notnull
+    {
+        if (x.Keys.Count != y.Keys.Count)
+        {
+            return false;
+        }
+
+        foreach (var key in x.Keys)
```

Comment:
Something to A/B test as I don't know the answer.

Would iterating like `foreach ((var key, var valueX) in x)` and avoiding the `x[key]` call be better or worse? I guess it depends on if iterating the keys is cheaper that iterating both and how often the right hand side of the `||` is accessed.